### PR TITLE
Implement ``drop_duplicates`` and ``unique``

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

File: dask_expr/reductions.py [hunk 1]
```diff
@@ -118,6 +118,67 @@ def _divisions(self):
...
+    def _subset_kwargs(self):
+        if is_series_like(self.frame._meta):
+            return {}
+        return {"subset": self.subset}
+
+    @property
+    def chunk_kwargs(self):
+        return {"ignore_index": self.ignore_index, **self._subset_kwargs()}
+
+    def _simplify_up(self, parent):
```
Comment (phofl) [inline dask_expr/reductions.py:178]: We can simplify here in some cases, but would prefer to do as follow up

File: dask_expr/tests/test_collection.py [hunk 2]
```diff
@@ -456,3 +457,29 @@ def test_repartition_divisions(df, opt):
...
             assert part.max() < df2.divisions[p + 1]
+
+
+def test_drop_duplicates(df, pdf):
+    assert_eq(df.drop_duplicates(), pdf.drop_duplicates())
+    assert_eq(
+        df.drop_duplicates(ignore_index=True), pdf.drop_duplicates(ignore_index=True)
+    )
+    assert_eq(df.drop_duplicates(subset=["x"]), pdf.drop_duplicates(subset=["x"]))
+    assert_eq(df.x.drop_duplicates(), pdf.x.drop_duplicates())
```
Comment (mrocklin) [inline dask_expr/tests/test_collection.py:468]: For tests like this I wonder if we should be relying on one giant parametrized test.

A lot of our tests are just `assert_eq(func(df), fund(pdf))`.  Maybe there is something we should be doing to make this easier / lighter weight?

File: dask_expr/expr.py [hunk 3]
```diff
@@ -404,6 +404,18 @@ def min(self, skipna=True, numeric_only=False, min_count=0):
...
+        # let pandas raise errors, e.g. not defined on DataFrame
+        self._meta.unique()
+        return Unique(self)
+
+    def drop_duplicates(self, subset=None, ignore_index=False):
+        if subset is not None:
+            # Fail early if subset is not valid
+            meta_nonempty(self._meta).drop_duplicates(subset=subset)
+
+        return DropDuplicates(self, subset=subset, ignore_index=ignore_index)
```
Comment (mrocklin) [inline dask_expr/expr.py:417]: If these are only on `Series/DataFrame` then maybe the answer is that we don't put them on `Expr`, but instead only put them on `Series/DataFrame`.

See https://github.com/mrocklin/dask-expr/issues/98#issuecomment-1556191825 for this same suggestion on a larger issue.

Comment (mrocklin) [reply dask_expr/expr.py:417]: Ah, I forgot that we already had this conversation on another PR as well.  Thoughts here?

File: dask_expr/tests/test_collection.py (hunk 2 above)
Comment (mrocklin) [reply dask_expr/tests/test_collection.py:468]: I'm also happy keeping these things separate, especially when there are lots of options like this to test.  Just a thought.

Comment (phofl) [reply dask_expr/tests/test_collection.py:468]: Generally, I prefer tests where the parametrisation isn't too complicated. Historically, we had some tests in pandas where the whole logic was encoded into the parametrisation, which made debugging a real PITA when something broke

File: dask_expr/expr.py (hunk 3 above)
Comment (phofl) [reply dask_expr/expr.py:417]: Yep makes sense, will update

Comment (phofl) [reply dask_expr/expr.py]: Updated

**Pull request merged**